persist/types.go

```diff
@@ -53,7 +53,7 @@ type Flush interface {
 	// Prepare prepares writing data for a given (shard, blockStart) combination,
 	// returning a PreparedPersist object and any error encountered during
 	// preparation if any.
-	Prepare(ns namespace.Metadata, shard uint32, blockStart time.Time) (PreparedPersist, error)
+	Prepare(ns namespace.Metadata, shard uint32, blockStart time.Time) (PreparedPersist, bool, error)
```

Comment:
would it be better to encode the exists semantics by returning a typed error?  

i.e. `Prepare(...)` returns `nil, ErrFilesetExists` when a file exists, so err != nil is sufficient to guarantee the returned `PreparedPersist` is valid.

Reply:
We could, but I think it leads to people not handling that error case properly. The signature I've defined here forces the caller to handle it or at least acknowledge that they've ignored that situation